packages/medusa/src/utils/authenticate-middleware.ts

```diff
@@ -67,7 +67,21 @@ export const authenticate = (
       }
     }
 
-    if (authUser) {
+    const isMedusaScope =
+      stringEqualsOrRegexMatch(authScope, "admin") ||
+      stringEqualsOrRegexMatch(authScope, "store")
+
+    const isRegistered =
+      !isMedusaScope ||
+      (authUser?.app_metadata?.user_id &&
+        stringEqualsOrRegexMatch(authScope, "admin")) ||
+      (authUser?.app_metadata?.customer_id &&
+        stringEqualsOrRegexMatch(authScope, "store"))
+
+    if (
+      authUser &&
+      (isRegistered || (!isRegistered && options.allowUnregistered))
+    ) {
       req.auth_user = {
```

Comment:
`req.auth_user` will likely change. See internal discussion.

Reply:
updated this pr to reflect internal discussion